homeassistant/components/imgw_pib/sensor.py

```diff
@@ -0,0 +1,101 @@
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ImgwPibConfigEntry
+from .const import ATTRIBUTION
+from .coordinator import ImgwPibDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class ImgwPibSensorEntityDescription(SensorEntityDescription):
+    """IMGW-PIB sensor entity description."""
+
+    value: Callable[[HydrologicalData], StateType]
+
+
+SENSOR_TYPES: tuple[ImgwPibSensorEntityDescription, ...] = (
+    ImgwPibSensorEntityDescription(
+        key="water_level",
+        translation_key="water_level",
+        native_unit_of_measurement=UnitOfLength.CENTIMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value=lambda data: data.water_level.value,
+    ),
+    ImgwPibSensorEntityDescription(
+        key="water_temperature",
+        translation_key="water_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value=lambda data: data.water_temperature.value,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ImgwPibConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a IMGW-PIB sensor entity from a config_entry."""
+    coordinator: ImgwPibDataUpdateCoordinator = entry.runtime_data.coordinator
+
+    async_add_entities(
+        [
+            ImgwPibSensorEntity(coordinator, description)
+            for description in SENSOR_TYPES
+            if getattr(coordinator.data, description.key).value is not None
+        ]
+    )
+
+
+class ImgwPibSensorEntity(
+    CoordinatorEntity[ImgwPibDataUpdateCoordinator], SensorEntity
+):
+    """Define IMGW-PIB sensor entity."""
+
+    _attr_has_entity_name = True
+    entity_description: ImgwPibSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: ImgwPibDataUpdateCoordinator,
+        description: ImgwPibSensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self._attr_unique_id = f"{coordinator.station_id}_{description.key}"
+        self._attr_device_info = coordinator.device_info
+        self._attr_attribution = ATTRIBUTION
+        self._attr_native_value = description.value(coordinator.data)
+        self.entity_description = description
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_native_value = self.entity_description.value(self.coordinator.data)
+        self.async_write_ha_state()
```

Comment:
Maybe it makes sense to create a separate function for setting the `_attr` but since you only have one attribute you're setting, why not use the native_value property directly?

Reply:
You're right, we can just use the `native_value` property here.